src/store/modules/app/mutations.js

```diff
@@ -181,6 +181,9 @@ const mutations = {
   setDarkmode(state, value) {
     state.configuratorSettings.darkmodeEnabled = value;
   },
+  setOSKeyboardLayout(state, value) {
+    state.configuratorSettings.osKeyboardLayout = value;
```

Comment:
As you added a new configuratorSettings key osKeyboardLayout, but state.js doesn't know about it. You should add this to intialConfig and give it a default otherwise the UI is a bit wiggy and we need to rev the localStorage version.

Reply:
The default should be US english as that is the existing behavior.

Reply:
That's strange because I did define the default osKeyboardLayout value in the initialConfig: https://github.com/qmk/qmk_configurator/blob/0a235d32896d9ba1e869da22c0a217c1d439fa2d/src/store/modules/app/state.js#L19

Reply:
I finally got around to testing this PR and I had the same issue of `osKeyboardLayout` being unset initially.

Reply:
And of course right after I say that, I have a theory to test, which I tested and proved.

The problem here is that `initialConfig` is only used if there are no settings saved in Local Storage at all. If you open your development console in the browser, delete `configuratorSettings` from Local Storage, then refresh the page, it works as expected.

We probably need a check to see if there's an existing setting for `osKeyboardLayout` in `configuratorSettings`, in addition to checking if there are any settings saved at all.

Reply:
> And of course right after I say that, I have a theory to test, which I tested and proved.
> 
> The problem here is that `initialConfig` is only used if there are no settings saved in Local Storage at all. If you open your development console in the browser, delete `configuratorSettings` from Local Storage, then refresh the page, it works as expected.
> 
> We probably need a check to see if there's an existing setting for `osKeyboardLayout` in `configuratorSettings`, in addition to checking if there are any settings saved at all.

I managed to reproduce the problem and issue a fix in commit 566678a